cypress/e2e/functional/case-search-spec.cy.js

```diff
@@ -149,6 +149,48 @@ describe('Case search', () => {
     cy.get('#case_number').type('TOO_MANY_RESULTS');
     cy.get('button').contains('Search').click();
 
-    cy.get('#search-results').should('contain', 'There are more than 500 results');
+    cy.get('app-search-error').should('contain', 'There are more than 500 results');
+  });
+
+  it('restore form values from previous search', () => {
+    cy.contains('Search').click();
+    cy.get('h1').should('contain', 'Search for a case');
+    cy.contains('Advanced search').click();
+    cy.get('#case_number').type('C20220620001');
+    cy.get('#courthouse').type('Cardiff');
+    cy.get('#courtroom').type('2');
+    cy.get('#specific-date-radio').click();
+    cy.get('#specific_date').type('03/07/2021');
+    cy.get('#defendant').type('Dean');
+    cy.get('#judge').type('Judge Dredd');
+    cy.get('#keywords').type('Daffy duck');
+    cy.get('button').contains('Search').click();
+
+    cy.get('#search-results').should('contain', '1 result');
+
+    // >> Click into a case
+    cy.get('[data-test-id=case-link]').click();
+
+    // Check case file page
+    cy.get('h1').should('contain', 'C20220620001');
+
+    // << Go Back
+    cy.get('.govuk-back-link').click();
+
+    // Check for Search page
+    cy.get('h1').should('contain', 'Search for a case');
+
+    cy.contains('Advanced search').click();
```

Comment:
I think advanced search should be open if any of the form values are populated

Reply:
Agreed 👍